opencog/persist/load_scm/fast_load.cpp

```diff
@@ -0,0 +1,172 @@
+#ifndef FAST_LOAD_CPP
+#define FAST_LOAD_CPP
+#include <opencog/atomspace/AtomSpace.h>
+#include <opencog/guile/SchemeEval.h>
+#include <opencog/atoms/atom_types/NameServer.h>
+#include "fast_load.h"
+
+using namespace opencog;
+
+#include <string>
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <stdexcept>
+
+bool check = false;
+void get_next_expr(const std::string& s, int& l, int& r)
+{
+    int l1 = l;
+    while(s[l1] != '(' && l1 < r) {
+        if(s[l1] != ' ' && s[l1] != '\t' && s[l1] != '\n') {
+            throw std::runtime_error("Invalid syntax #1 in " + s.substr(l,r-l+1) + " at |" + s[l1] + "|");
```

Comment:
it would be more efficient (fewer compares) to write this as
```
while (s[l1] == ' ' or s[l1] == '\t' or s[l1] == '\n') l1++;
if (s[l1] != '(') throw ...
````